### ISSUE TITLE
Setting unknown property: barrelstrength\sproutemail\events\notificationevents\EntriesSave::userGroupIds

### ISSUE THREAD

**Issue by BenParizek** [barrelstrength/craft-sprout-email#121]
A user writes:

> I have an email configured to send when a new entry is created. This is using the Guest Entries plugin, not sprout forms.
> I get this error message when I save the entry:
> Setting unknown property: barrelstrength\sproutemail\events\notificationevents\EntriesSave::userGroupIds
> 
> I’m also unable to edit the notification settings as it gives me the same error message.
> 
> I’ve looked in the database and the settings for this notification in craft_sproutemail_notificationemails are:
> {"whenNew":"","whenUpdated":"","userGroupIds":"*”}
> 
> I removed the userGroupIds to leave me just this:
> {"whenNew":"","whenUpdated":""}
> 
> I was then able to edit the notification but the fields for “When entry is created?” and “When entry is updated?” weren’t checked.
> I ticked them and this time chose the specific section instead of all and the setting in the database had changed to this:
> {"whenNew":"1","whenUpdated":"1","sectionIds":["12"]}
> 
> I’m not getting error messages any more but I’m also not receiving an email. That might be linked to my test setup/docker configuration so I’ll work on checking that.
> One other point, I see this is using a normal sectionId of 12, the new version of Guest Entries plugin is forcing me to use the new uid identifier for the section. I don’t think that’s the issue but I’m going to try to switch it back to all sections next.

<img src="https://frontapp.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_qflv)

**Comment by BenParizek** [comment]
Related https://github.com/barrelstrength/craft-sprout-email/issues/125

**Comment by admench** [comment]
Seeing the same thing here. Craft 3.4.9

```
barrelstrength/sprout-base               5.1.2    Common libraries for Sprout Plugins for Craft CMS
barrelstrength/sprout-base-email         1.2.3    Common email libraries for Sprout Plugins for Craft CMS
barrelstrength/sprout-base-fields        1.3.3    Common fields libraries for Sprout Plugins for Craft CMS
barrelstrength/sprout-base-import        1.0.6    Common import libraries for Sprout Plugins for Craft CMS
barrelstrength/sprout-base-lists         1.0.5    Common lists libraries for Sprout Plugins for Craft CMS
barrelstrength/sprout-base-reports       1.4.2    Common reports libraries for Sprout Plugins for Craft CMS
barrelstrength/sprout-email              4.1.4    Flexible, integrated email marketing and notifications.
barrelstrength/sprout-fields             3.6.1    International, Craft-friendly field types.
barrelstrength/sprout-forms              3.8.5    Simple, beautiful forms. 100% control.
barrelstrength/sprout-lists              2.0.1    Subscribe and unsubscribe to things.
barrelstrength/sprout-reports            1.3.2    Powerful custom reports.
```

```
yii\base\UnknownPropertyException: Setting unknown property: barrelstrength\sproutemail\events\notificationevents\EntriesSave::userGroupIds in /Users/username/code/sites/sites/vendor/yiisoft/yii2/base/Component.php:209
Stack trace:
#0 /Users/username/code/sites/sites/vendor/yiisoft/yii2/BaseYii.php(558): yii\base\Component->__set('userGroupIds', '*')
#1 /Users/username/code/sites/sites/vendor/yiisoft/yii2/base/BaseObject.php(107): yii\BaseYii::configure(Object(barrelstrength\sproutemail\events\notificationevents\EntriesSave), Array)
#2 /Users/username/code/sites/sites/vendor/barrelstrength/sprout-base-email/src/services/NotificationEmailEvents.php(276): yii\base\BaseObject->__construct(Array)
#3 /Users/username/code/sites/sites/vendor/barrelstrength/sprout-base-email/src/controllers/NotificationsController.php(180): barrelstrength\sproutbaseemail\services\NotificationEmailEvents->getNotificationEmailEvents(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail))
#4 [internal function]: barrelstrength\sproutbaseemail\controllers\NotificationsController->actionEditNotificationEmailTemplate('global', '242', Object(barrelstrength\sproutbaseemail\elements\NotificationEmail))
#5 /Users/username/code/sites/sites/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#6 /Users/username/code/sites/sites/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#7 /Users/username/code/sites/sites/vendor/craftcms/cms/src/web/Controller.php(178): yii\base\Controller->runAction('edit-notificati...', Array)
#8 /Users/username/code/sites/sites/vendor/yiisoft/yii2/base/Module.php(528): craft\web\Controller->runAction('edit-notificati...', Array)
#9 /Users/username/code/sites/sites/vendor/craftcms/cms/src/web/Application.php(291): yii\base\Module->runAction('sprout-base-ema...', Array)
#10 /Users/username/code/sites/sites/vendor/yiisoft/yii2/web/Application.php(103): craft\web\Application->runAction('sprout-base-ema...', Array)
#11 /Users/username/code/sites/sites/vendor/craftcms/cms/src/web/Application.php(276): yii\web\Application->handleRequest(Object(craft\web\Request))
#12 /Users/username/code/sites/sites/vendor/yiisoft/yii2/base/Application.php(386): craft\web\Application->handleRequest(Object(craft\web\Request))
#13 /Users/username/code/sites/sites/web/index.php(22): yii\base\Application->run()
#14 /Users/username/.composer/vendor/laravel/valet/server.php(158): require('/Users/adammenc...')
#15 {main}
```

I can't visit pages such as `admin/sprout-email/notifications/edit/242`
If you need any extra info from me I'll happily provide it.

**Comment by BenParizek** [comment]
@admench I'm working on getting a migration in place to correct this but I'm also in the middle of a couple of busy weeks.

If you need to get this resolved before I get a fix out, I believe the issue is that we stored settings for Notification Events even if you changed the settings in the version you upgraded from. This could mean that there were two sets of settings stored for a Notification Email in the database but we only used the active one when sending the notification. This detail is not taken into account in our migration right now, so you can end up with a Notification Email with the wrong settings and when Sprout tries to resolve those settings throws the error your seeing above.

The fix we get out will require re-updating your database. If you don't want to roll back or don't want to wait you can go to the Notification Email settings table in the db (`sproutemail_notificationemails.settings`), set the settings column to NULL for the Notification causing the issue, and that should get your page loading again. You will then have to add your settings back to the Notification Email.

**Comment by admench** [comment]
Fantastic, I'll patch that column nullable value setting now. Thank you

**Comment by BenParizek** [comment]
@admench I expect you've resolved this by now but just in case, we've just released Sprout Email v4.1.6 which updates the migration to resolve this issue.

**Comment by admench** [comment]
Works a treat by manually nulling those notification's settings column. I also updated to latest. Many thanks
